core/shared/src/main/scala/scalaz/zio/Task.scala

```diff
@@ -0,0 +1,341 @@
+
+  /** See [[scalaz.zio.ZIO.fail]]
+   */
+  final def fail(error: Throwable): IO[Throwable, Nothing] = ZIO.fail(error)
+
+  /** See [[scalaz.zio.ZIO.flatten]]
+   */
+  final def flatten[A](zio: Task[Task[A]]): Task[A] =
+    ZIO.flatten(zio)
+
+  /** See [[scalaz.zio.ZIO.foldLeft]]
+   */
+  final def foldLeft[S, A](in: Iterable[A])(zero: S)(f: (S, A) => Task[S]): Task[S] =
+    ZIO.foldLeft(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.foreach]]
+   */
+  final def foreach[A, B](in: Iterable[A])(f: A => Task[B]): Task[List[B]] =
+    ZIO.foreach(in)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachPar]]
+   */
+  final def foreachPar[A, B](as: Iterable[A])(fn: A => Task[B]): Task[List[B]] =
+    ZIO.foreachPar(as)(fn)
+
+  /** See [[scalaz.zio.ZIO.foreachParN]]
+   */
+  final def foreachParN[A, B](n: Long)(as: Iterable[A])(fn: A => Task[B]): Task[List[B]] =
+    ZIO.foreachParN(n)(as)(fn)
+
+  /** See [[scalaz.zio.ZIO.foreach_]]
+   */
+  final def foreach_[A](as: Iterable[A])(f: A => Task[_]): Task[Unit] =
+    ZIO.foreach_(as)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachPar_]]
+   */
+  final def foreachPar_[A, B](as: Iterable[A])(f: A => Task[_]): Task[Unit] =
+    ZIO.foreachPar_(as)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachParN_]]
+   */
+  final def foreachParN_[A, B](n: Long)(as: Iterable[A])(f: A => Task[_]): Task[Unit] =
+    ZIO.foreachParN_(n)(as)(f)
+
+  /** See [[scalaz.zio.ZIO.forkAll]]
+   */
+  final def forkAll[A](as: Iterable[Task[A]]): UIO[Fiber[Throwable, List[A]]] =
+    ZIO.forkAll(as)
+
+  /** See [[scalaz.zio.ZIO.forkAll_]]
+   */
+  final def forkAll_[A](as: Iterable[Task[A]]): UIO[Unit] =
+    ZIO.forkAll_(as)
+
+  /** See [[scalaz.zio.ZIO.fromFunction]]
+   */
+  final def fromFunction[A](f: Any => A): UIO[A] =
+    ZIO.fromFunction(f)
+
+  /** See [[scalaz.zio.ZIO.fromFunctionM]]
+   */
+  final def fromFunctionM[A](f: Any => IO[Throwable, A]): Task[A] =
+    ZIO.fromFunctionM(f)
+
+  /** See [[scalaz.zio.ZIO.fromEither]]
+   */
+  final def fromEither[A](v: => Either[Throwable, A]): IO[Throwable, A] =
+    ZIO.fromEither(v)
+
+  /** See [[scalaz.zio.ZIO.fromFiber]]
+   */
+  final def fromFiber[A](fiber: => Fiber[Throwable, A]): IO[Throwable, A] =
+    ZIO.fromFiber(fiber)
+
+  /** See [[scalaz.zio.ZIO.fromFiberM]]
+   */
+  final def fromFiberM[A](fiber: IO[Throwable, Fiber[Throwable, A]]): IO[Throwable, A] =
+    ZIO.fromFiberM(fiber)
+
+  /** See [[scalaz.zio.ZIO.fromFuture]]
+   */
+  final def fromFuture[A](make: ExecutionContext => scala.concurrent.Future[A]): Task[A] =
+    ZIO.fromFuture(make)
+
+  /** See [[scalaz.zio.ZIO.fromTry]]
+   */
+  final def fromTry[A](value: => scala.util.Try[A]): Task[A] =
+    ZIO.fromTry(value)
+
+  /** See [[scalaz.zio.ZIO.halt]]
+   */
+  final def halt(cause: Cause[Throwable]): IO[Throwable, Nothing] = ZIO.halt(cause)
+
+  /** See [[scalaz.zio.ZIO.identity]]
+   */
+  final def identity: UIO[Any] = ZIO.identity
```

Comment:
This method only makes sense when `R` != `Any`, and should be deleted from `Task`.